### ISSUE TITLE
Startup mode Maximized with none decorations cause fullscreen on Windows

### ISSUE THREAD

**Issue by codehz** [alacritty/alacritty#3784]
For bug reports, the following information can help speed up the process. Please describe the bug that you have found and what you would expect to happen instead.

### System

OS: Windows
Version: v0.4.3-rc1
Windows: ConPTY (also for WinPTY)

### Logs

```
[2020-05-28 21:51:14.808682000] [INFO] Welcome to Alacritty
[2020-05-28 21:51:14.808682000] [INFO] Configuration loaded from "C:\Users\CodeHz\AppData\Roaming\alacritty\alacritty.yml"
[2020-05-28 21:51:14.880889000] [INFO] Device pixel ratio: 1.25
[2020-05-28 21:51:14.889035200] [INFO] Initializing glyph cache...
[2020-05-28 21:51:14.913584800] [INFO] ... finished initializing glyph cache in 0.0243273s
[2020-05-28 21:51:14.914603200] [INFO] Cell Size: 10 x 25
[2020-05-28 21:51:14.921805400] [INFO] Padding: 2 x 9
[2020-05-28 21:51:14.921805400] [INFO] Width: 1024, Height: 768
[2020-05-28 21:51:14.953070100] [INFO] PTY Dimensions: Line(30) x Column(102)
[2020-05-28 21:51:14.953070100] [INFO] Using WinPTY backend
[2020-05-28 21:51:15.095485000] [INFO] Initialisation complete
[2020-05-28 21:51:15.111089300] [INFO] Width: 1920, Height: 1080
```

**Comment by kchibisov** [comment]
And how it looks with decorations? Also, does `Alt + Enter` unfullscreens it or it just does nothing on a first press? I don't have Windows system to test how maximize works on it.

**Comment by codehz** [comment]
with decorations: normal maximized
Alt + Enter **twice** : normal maximized  (so the first `alt + enter` is switch to _real fullscreen mode_

**Comment by kchibisov** [comment]
What are you expecting from maximized window **without** decorations?  I'll assume it should just look like a fullscreen window, and from what you describe it's just a normal window, but big.

>so the first alt + enter is switch to real fullscreen mode

it's not fullscreen window. Could you show me a screenshot of normal maximized window? I'll assume that it doesn't cover dock/statusbar, right?

**Comment by codehz** [comment]
> Could you show me a screenshot of normal maximized window?

![image](https://user-images.githubusercontent.com/13158903/83151830-a1e69700-a12f-11ea-8b04-bd53250c7255.png)

that's expected

but actually 

![image](https://user-images.githubusercontent.com/13158903/83151596-57fdb100-a12f-11ea-8593-3c26b49fb6da.png)

(combine with another bug that the vim cannot recognize the size of maximized window

**Comment by kchibisov** [comment]
>(combine with another bug that the vim cannot recognize the size of maximized window before resize

It's a known vim/nvim bug (not our bug). It can't handle multiple resizes during starturp due to some "reasons".

**Comment by codehz** [comment]
> It's a known vim/nvim bug (not our bug)

nope, it also happened to bash (after I try to execute more than N command that will exceed to the limit

**Comment by codehz** [comment]
> that's expected (after alt-enter twice)

(nope, the last line is behind of the taskbar...

**Comment by chrisduerr** [comment]
> the last line is behind of the taskbar

So it's not actually fullscreen, but it's just too big?

**Comment by codehz** [comment]
> but it's just too big

yes, just like some game's _seamless window mode_

**Comment by chrisduerr** [comment]
Reported upstream in https://github.com/rust-windowing/winit/issues/1582.

**Comment by tinnick** [comment]
Hey guys, I have the same problem.

Just so you know it's not only this man.

**Comment by zzantares** [comment]
This one appears to be solved (at least for me) I can see it doesn't start in full screen and the task bar isn't hidden anymore:

![Image](https://github.com/user-attachments/assets/1e3c4e06-6f3d-4136-8e13-8ce354aeae2a)

**Comment by chrisduerr** [comment]
Closing since nobody seems to be complaining about this issue still being present. If it still exists please let me know and I'll reopen.
